mountpoint-s3/src/upload.rs

```diff
@@ -49,9 +57,17 @@ impl<Client: ObjectClient> Uploader<Client> {
         &self,
         bucket: &str,
         key: &str,
-    ) -> ObjectClientResult<UploadRequest<Client>, PutObjectError, Client::ClientError> {
+    ) -> Result<UploadRequest<Client>, UploadPutError<PutObjectError, Client::ClientError>> {
         UploadRequest::new(Arc::clone(&self.inner), bucket, key).await
     }
+
+    #[cfg(test)]
+    pub fn corrupt_sse(&mut self, sse_type: Option<String>, sse_kms_key_id: Option<String>) {
```

Comment:
not sure if we want failure-injection api at the Uploader level; if not we can omit the `test_open_with_corrupted_sse` test

Reply:
Yeah, this one we probably don't need the method, since you can write the test in the same module and just touch the private fields directly.

Reply:
Oh I see, each test uses the method from the other file. Seems fine then.